bin/gwd/request.ml

```diff
@@ -552,6 +552,24 @@ let treat_request =
           w_wizard @@ w_base @@ UpdateInd.print_del
         | "DEL_IND_OK" ->
           w_wizard @@ w_lock @@ w_base @@ UpdateIndOk.print_del
+        | "DOC" ->
+          w_base @@ fun conf base -> (
+          match Util.p_getenv conf.env "s" with
+          | Some f ->
+                if Filename.check_suffix f ".txt" then
+                  let f = Filename.chop_suffix f ".txt" in
+                  SrcfileDisplay.print_source conf base f
+                else ImageDisplay.print_source conf f
+          | _ -> incorrect_request conf base)
+        | "DOCH" ->
+          w_base @@ fun conf base -> (
+          match Util.p_getenv conf.env "s" with
+          | Some f ->
+                if Filename.check_suffix f ".txt" then
+                  let f = Filename.chop_suffix f ".txt" in
+                  SrcfileDisplay.print_source conf base f
+                else ImageDisplay.print_html conf
+          | _ -> incorrect_request conf base)
```

Comment:
We can factorize DOC and DOCH in a single function that take the printing function (print_source/print_html) as parameter

Reply:
Oui, mais print_html n'attend pas de paramètre alors que print_source attend f !!
Pour print_html, le nom du fichier est dans l'url (s=file) et est passé à la commande m=IM
Comment on fait?

Reply:
ah mince,

On doit pouvoir s'en sortir en faisant quelque chose comme :

```ocaml
let doc_aux conf base print = 
          w_base @@ fun conf base -> (
          match Util.p_getenv conf.env "s" with
          | Some f ->
                if Filename.check_suffix f ".txt" then
                  let f = Filename.chop_suffix f ".txt" in
                  SrcfileDisplay.print_source conf base f
                else print conf f
          | _ -> incorrect_request conf base)
```

```ocaml
| "DOC" -> doc_aux conf base ImageDisplay.print_source
| "DOCH" -> doc_aux conf base (fun conf base _f -> ImageDisplay.print_html conf base)
```